engine/Assets/Scripts/UI/Dynamic/Panels/Configuring/Scoring/ZoneConfigPanel.cs

```diff
@@ -0,0 +1,315 @@
+            .ApplyTemplate(VerticalLayout).AddOnValueChangedEvent(
+                (s, v) => {
+                    _data.ZScale = v;
+                    DataUpdated();
+                });
+
+        GameObject obj;
+        if (_zone is null) {
+            obj = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            _zone = new ScoringZone(obj, "temp scoring zone", Alliance.Blue, 0, false);
+        } else {
+            obj = _zone.GameObject;
+            UseZone(_zone);
+        }
+
+        GizmoManager.SpawnGizmo(obj.transform,
+            t => {
+                obj.transform.position = t.Position;
+                obj.transform.rotation = t.Rotation;
+            },
+            t => { });
+
+        return true;
+    }
+
+    private void CopyDataToZone(ScoringZoneData data, ScoringZone zone) {
+        zone.Name = _zoneNameInput.Value;
+        zone.GameObject.name = data.Name;
+        zone.GameObject.tag = data.Alliance == Alliance.Red ? "red zone" : "blue zone";
+        zone.GameObject.transform.parent = data.Parent;
+        zone.Alliance = data.Alliance;
+        zone.Points = data.Points;
+        zone.GameObject.transform.localScale = new Vector3(data.XScale, data.YScale, data.ZScale);
+        zone.DestroyObject = data.DestroyGamepiece;
+    }
+
+    private void ConfigureAllianceButton() {
+        if (_data.Alliance == Alliance.Blue) {
+            _zoneAllianceButton.StepIntoLabel(l => l.SetText("Blue Alliance")).SetBackgroundColor<Button>(Color.blue);
+        } else {
+            _zoneAllianceButton.StepIntoLabel(l => l.SetText("Red Alliance")).SetBackgroundColor<Button>(Color.red);
+        }
+    }
+
+    public void UseZone(ScoringZone zone) {
+        _zone = zone;
+        _data.Name = _zone.Name;
+        _zoneNameInput.SetValue(_zone.Name);
+
+        _data.Alliance = zone.Alliance;
+        ConfigureAllianceButton();
+
+        _data.Points = zone.Points;
+        _pointsInputField.SetValue(_zone.Points);
+        
+        _data.DestroyGamepiece = zone.DestroyObject;
+        _deleteGamepieceToggle.SetState(_zone.DestroyObject, notify: false);
+        
+        var localScale = zone.GameObject.transform.localScale;
+        _data.XScale = localScale.x;
+        _xScaleSlider.SetValue(_data.XScale);
+        _data.YScale = localScale.y;
+        _yScaleSlider.SetValue(_data.YScale);
+        _data.ZScale = localScale.z;
+        _zScaleSlider.SetValue(_data.ZScale);
+    }
+
+    private void DataUpdated() {
+        _zone.Alliance = _data.Alliance;
+        _zone.Points = _data.Points;
+        _zone.GameObject.transform.localScale = new Vector3(_data.XScale, _data.YScale, _data.ZScale);
+    }
+
+    public void SetCallback(Action<ScoringZone, bool> callback) {
+        _callback = callback;
+    }
+    
+    public void SelectParentButton(Button b) {
+        if (!_selectingNode) {
+            // I don't like this; do we just want all field rigidbodies to detect collisions?
```

Comment:
They're set to not detect collisions when freezing (FieldSimObject.cs line 46). I didn't look too much into the reason for this though, maybe it's unnecessary or just needs to be put back when you're done selecting? I did test specifically for this and the code you have now works.

Reply:
Makes sense. What part about the current code don't you like?

Reply:
I didn't like the whole keeping track of the initial collision-detecting state of all of the field elements because it seems inelegant, and I don't know enough about the field structure to know if there are any elements that don't detect collisions by default, or if it would be okay to set `detectCollisions` to true for all of them.

Reply:
Gotcha. I can't say I'm sure, but if freeze affects all except grounded and currently possessed, selecting parent probably can too.

Reply:
Ha just got your edit, sounds good.